ctapipe/image/cleaning.py

```diff
@@ -355,3 +355,43 @@ def fact_image_cleaning(
         geom, pixels_to_keep, arrival_times, min_number_neighbors, time_limit
     )
     return pixels_to_keep
+
+
+def biggest_island(geom, image, mask):
```

Comment:
I would recommend to make this smaller. Give this function just the labels and return a new mask with fewer `Trues`, thus this function gets smaller and is more general. Also  `num_islands` will be calculated twice if you also use it in the regular analysis.

So I would change this to:
```python
def largest_island(labels):
    return labels == np.argmax(np.bincount(labels[labels > 0]))  # no need for where
```

Reply:
Thanks for this input.
Let me just understand better:

- I don't understand the "fewer Trues" part, for me the only Trues are always those related to the largest island and the mask reflects always the original geometry,

- I don't understand why _number_of_islands_, should be called twice in the regular analysis (do you mean _protopipe_?)

- you suggest that in the regular analysis I call first _number_of_islands_, then _largest_island_ and finally I select the geometry and image instead of doing everything inside _biggest_island_,

Reply:
* Yes, this was a bit confusing, I meant return a mask where only the pixels in the largest group are `True`
* number_of_islands is a useful paramter for G/H separation, that should be stored into DL1.2, so if you call it inside largest_island without being able to get the number of islands, you need to call it again to be able to store the value.
* Yes, exactly. Small, composable functions!

Reply:
> * Yes, this was a bit confusing, I meant return a mask where only the pixels in the largest group are `True`

This was already happening in my function, am I wrong?  _biggest_mask_ is a mask with the same dimensions of the original camera, but only the pixels belonging to the biggest island are set to True (`biggest_mask = labels == biggest_label`).

> * number_of_islands is a useful paramter for G/H separation, that should be stored into DL1.2, so if you call it inside largest_island without being able to get the number of islands, you need to call it again to be able to store the value.

OK, I understand now.

> * Yes, exactly. Small, composable functions!

Yes, I agree.

Reply:
> I don't understand why number_of_islands, should be called twice in the regular analysis (do you mean protopipe?)

For example, we will include the number of islands in the DL1 parameters (so we should avoid computing it twice)